cmd/agent/app/check.go

```diff
@@ -144,7 +144,7 @@ var checkCmd = &cobra.Command{
 				continue
 			}
 
-			if check.IsJMXConfig(conf.Name, conf.InitConfig) {
+			if check.IsJMXConfig(conf) {
```

Comment:
I think this will require a more fine-grained approach, since this only tells us that one instance of the config is a JMXFetch config, not necessarily all, so the instances that are not JMXFetch configs should be go through the logic for "regular" checks below